.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,147 @@
+      # with the maximum checkout speed
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: ${{ github.event.pull_request.commits }}
+      - name: Prepare commits
+        id: prepare
+        run: |
+          git fetch --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+          COMMITS_TO_TEST=$(git rev-list origin/${{ github.base_ref }}..${{ github.head_ref }} --pretty='"%H"' --no-commit-header)
+          echo "matrix={ \"commit\": [${COMMITS_TO_TEST//$'\n'/, }] }" | tee -a ${GITHUB_OUTPUT}
+
+  regression-tests:
+    runs-on: [ self-hosted, ci-runner-compiler ]
+    container:
+      image: matterlabs/llvm_runner:ubuntu22-llvm15-latest
+      options: -m 110g
+    needs: prepare-commits
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix: ${{ fromJson(needs.prepare-commits.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.commit }}
+          fetch-depth: ${{ github.event.pull_request.commits }}
+          path: "llvm"
+
+      - name: Check formatting
+        working-directory: llvm
+        shell: bash
+        run: |
+          git fetch --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+          CHANGED_FILES="$(git diff --name-only origin/${{ github.base_ref }})"
+          CHANGED_FILES="${CHANGED_FILES//$'\n'/,}"
+          echo "Formatting files:"
+          echo "${CHANGED_FILES}"
+          python -m pip install -r ./llvm/utils/git/requirements_formatting_era_llvm.txt
+          python ./llvm/utils/git/code-format-helper-era-llvm.py \
+            --token ${{ secrets.GITHUB_TOKEN }} \
+            --issue-number ${{ github.event.pull_request.number }} \
+            --start-rev ${{ github.event.pull_request.base.sha }} \
+            --end-rev ${{ github.event.pull_request.head.sha }} \
+            --changed-files "${CHANGED_FILES}"
+
+      - name: Linting (clang-tidy)
+        working-directory: llvm
+        shell: bash
+        run: |
+          set -euxo pipefail
+          git fetch --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+          git diff -U0 origin/${{github.base_ref}}
+          git diff -U0 origin/${{github.base_ref}} \
+            | ./clang-tools-extra/clang-tidy/tool/clang-tidy-diff_Zegar.py \
+              -p1 -clang-tidy-binary $(which clang-tidy) \
+              -path ../target-llvm/build-final/compile_commands.json
+
+      - name: Build LLVM
+        uses: ./llvm/.github/actions/build
+        with:
+          extra-args: "\\-DLLVM_ENABLE_WERROR=On \\-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+          enable-tests: true
+
+      - name: Running Lit tests with default options
+        run: |
+          BINDIR=$(target-llvm/build-final/bin/llvm-config --bindir)
+          ${BINDIR}/llvm-lit -vv -s './target-llvm/build-final/test'
```

Comment:
> _This loses --xunit-xml-output test-results-default.xml_

Yeah, I removed this, because right now we check too many commits and each of them produce 3 different XML files. Additionally, with each interactive rebase/force push commits SHA will be new, and these checks will continue to create new messages in PRs. I made a decision to remove them to not flood in PRs.

Reply:
> Until [recently](https://github.com/matter-labs/era-compiler-llvm/pull/0e65463191b6a500a2a9b52991#diff-7f9aa370a2a8bbd346daab0b4359cf1f418c1a425a1fe143aaf0d14a7e7a2270R77) this was instead the ninja target check-llvm; the next step below still specifies the test directory in the source hierarchy: './llvm/llvm/test/CodeGen/EraVM/'. I hadn’t been aware that our build-final has a copy of the tests source. (This might matter when using the Google Unit Tests, which generate some files.)

`check-llvm` does exacly this call to `llvm-lit` that I'm using here. But probably, it would be better to use `check-llvm` here as we do not specify `--xunit-xml-output` option anymore. Code updated.

Reply:
I hadn’t realized that; I’ve generally called `llvm-lit` with an explicit path.  BeyondCompare tells me that my (somewhat old) `./target-llvm/build-final/test` is missing some test files that `llvm/llvm/test` has.  First alphabetically:  

`/Users/flash/Documents/Code/compiler-tester/llvm/llvm/test/Analysis/AliasSet/argmemonly.ll` etc.

```
flash@Astrophage llvm % ls /Users/flash/Documents/Code/compiler-tester/target-llvm/build-final/test/Analysis/AliasSet
Output
```

Reply:
I've opened [CPR-1648](https://linear.app/matterlabs/issue/CPR-1648/investigate-difference-in-test-directories-for-llvm-check) for that to not block this PR.